src/main/java/com/puppycrawl/tools/checkstyle/utils/XpathUtil.java

```diff
@@ -157,6 +159,7 @@ public static String getTextAttributeValue(DetailAST ast) {
         if (ast.getType() == TokenTypes.STRING_LITERAL) {
             text = text.substring(1, text.length() - 1);
         }
+        text = Pattern.compile("[\n]+").matcher(text).replaceAll("\\\\n");
```

Comment:
Do "compile" and save result in field to do do this ones for whole run of checkstyle

Reply:
I was not sure how to do this, so I had changed it at `GenerateXpathQuery.generateXpathQuery()` since it is nearest string from main return function that can modified. Please suggest changes.

Reply:
Make field `ptivate Pattern endlunes = Pattern.compile("[\n]+")`

At this line do 
`endlines.matches`

Reply:
Ok, thanks. I wrongly miss interpreted it as this change should occur after creating of total string of Xpath. That paid off by chance and now only once this encoding happens for a element. Also, changes suggested by you are done.

Reply:
Done